File: backend/src/main/java/com/campus/Campus_Connect/features/map/dto/response/EventPreviewResponse.java
package com.campus.Campus_Connect.features.map.dto.response;

import com.campus.Campus_Connect.features.event.entity.enums.RegistrationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.time.Instant;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EventPreviewResponse {

    private Integer id;

    private String title;

    private String description;

    private String posterUrl;

    private Instant startTime;

    private Instant endTime;

    private String venue;

    private RegistrationType registrationType;

    private String registrationLink;

    private Boolean isJoined;

    private Boolean isReminderEnabled;

    private Integer priority;

    private List<HostPreview> hosts;


    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class HostPreview {

        private Integer userId;

        private String fullName;

        private String avatarUrl;

        private String role;
    }
}

File: android-app/app/src/main/java/com/example/campusconnect/feature/map/data/remote/response/EventPreviewResponse.kt
package com.example.campusconnect.feature.map.data.remote.response

import com.example.campusconnect.feature.map.model.HostInfo
import com.example.campusconnect.feature.map.model.MapEventInfo
import com.google.gson.annotations.SerializedName

data class EventPreviewResponse(
    val id: Int,
    val title: String,
    val description: String? = null,
    val posterUrl: String? = null,
    val startTime: String? = null,
    val endTime: String? = null,
    val venue: String? = null,
    val registrationType: String? = null,
    val registrationLink: String? = null,
    val isJoined: Boolean? = null,
    val isReminderEnabled: Boolean? = null,
    val priority: Int? = null,
    val hosts: List<EventHostResponse>? = null
)

fun EventPreviewResponse.toMapEventInfo(): MapEventInfo {
    val hostList = hosts?.map { host ->
        HostInfo(
            id = host.userId,
            name = host.fullName,
            avatarUrl = host.avatarUrl
        )
    } ?: emptyList()

    return MapEventInfo(
        id = id,
        title = title,
        hostName = hostList.firstOrNull()?.name ?: "Campus Team",
        date = startTime ?: "TBA",
        time = if (startTime != null && endTime != null) {
            "$startTime - $endTime"
        } else {
            "TBA"
        },
        description = description ?: "",
        venue = venue ?: "Campus Complex",
        posterUrl = posterUrl,
        posterResId = null,
        hosts = hostList,
        isJoined = isJoined ?: false
    )
}

